tests/Microsoft.Identity.Test.Integration.Win8/DeviceAuthenticationTests.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Threading;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Json.Linq;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Integration.net45.HeadlessTests
+{
+    [TestClass]
+    public class DeviceAuthenticationTests
+    {
+        private const string _claims = "{\"access_token\":{\"deviceid\":{\"essential\":true}}}";
+        private const string _publicClientId = "4b0db8c2-9f26-4417-8bde-3f0e3656f8e0";
+        private const string _deviceAuthuser = "idlabca@msidlab8.onmicrosoft.com";
+
+        [TestMethod]
+
+        public void PKeyAuthNonInteractiveTest()
+        {
+            //Arrange
+            var msalPublicClient = PublicClientApplicationBuilder
+                .Create(_publicClientId)
+                .Build();
+
+            //Act
+            var authResult = msalPublicClient.AcquireTokenByUsernamePassword(new[] { "user.read" }, _deviceAuthuser, new NetworkCredential("", LabUserHelper.FetchUserPassword("msidlab8")).SecurePassword)
+            .WithClaims(JObject.Parse(_claims).ToString())
+            .ExecuteAsync(CancellationToken.None).Result;
+
+            //Assert
```

Comment:
How do we know this test does anything? I don't see any assertion that indicates that device auth was performed. 

One idea would be to use https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/b84322c945e0b8e5f3e237ea9b9e695b50fd1a36/tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/HttpSnifferClientFactory.cs#L11 to interpret the traffic